### DISCUSSION TITLE
Keycloak Integration

### DISCUSSION THREAD

**Discussion by JGui-Fernandes** [drakkan/sftpgo#1461] [Q&A]
Hello,
I'm currently working on a Keycloak integration project with the sftpgo platform, both hosted in a Kubernetes Cluster. I'm facing some challenges with the configurations.
Specifically, I'm having difficulty understanding the OIDC documentation. I've successfully created the Realm and configured it accordingly. Additionally, I've set up the Client, providing my URLs: "Valid redirect URIs: http://sftpgo.com/*" and "Config URL: https://keycloak.com/auth/realms/sftpgo" (as specified in sftpgo.json).
However, after completing these steps, when attempting to authenticate in the sftpgo platform, it only opens the regular user interface. I'm wondering how I can configure Keycloak to properly redirect my user to the correct URL within the Kubernetes environment.
Any assistance on resolving this redirection issue would be greatly appreciated.
Thank you.

**Comment by drakkan** [comment]
Hello, once you properly configure OpenID connect you should see a "Login with OpenID" button

**Reply by JGui-Fernandes** [reply]
Hello, thanks for answering...
It didn't happen. The Sftpgo page is exactly the same as it was.
Maybe the problem is in "config_url" in sftpgo.json, but i'm not sure.

**Reply by marsteel** [reply]
make sure the OIDC setting are passed to sftpgo via ENV
- SFTPGO_HTTPD__BINDINGS__OIDC__CLIENT_ID="sftpgo-client"
- SFTPGO_HTTPD__BINDINGS__OIDC__CLIENT_SECRET="aAl4DVwhquXCX2LDKL0J9O0dVMXEo1Ga"
- SFTPGO_HTTPD__BINDINGS__OIDC__CONFIG_URL=http://keycloak:10991/auth/realms/sftpgo
- SFTPGO_HTTPD__BINDINGS__OIDC__REDIRECT_BASE_URL=http://sftpgo:10200/
- SFTPGO_HTTPD__BINDINGS__OIDC__USERNAME_FIELD="preferred_username",
- SFTPGO_HTTPD__BINDINGS__OIDC__ROLE_FIELD="sftpgo_role"
